src/app/api/notification/notification.service.ts

```diff
@@ -49,16 +49,71 @@ export class NotificationService extends BaseService {
             email: getEmailDetails(actionUserName, task.title)[action],
           },
         }
-
         return copilotUtils.createNotification(notificationDetails)
       })
 
-      await Promise.all(promises)
+      return await Promise.all(promises)
     } catch (error) {
       console.error(`Failed to send notifications for action: ${action}`, error)
     }
   }
 
+  /**
+   * Adds a ClientNotification column associated with a task
+   * @param task Associated task
+   * @param notification Associated notification
+   * @returns New ClientNotification object
+   */
+  async addToClientNotifications(task: Task, notification: NotificationCreatedResponse): Promise<ClientNotification> {
+    return await this.db.clientNotification.create({
+      data: {
+        clientId: z.string().parse(notification.recipientId),
+        notificationId: notification.id,
+        taskId: task.id,
+      },
+    })
+  }
+
+  /**
+   * Marks a client notification in Copilot Notifications service as read
+   * @param id Notification ID for object as exists in Copilot
+   */
+  async markClientNotificationAsRead(task: Task) {
+    const copilot = new CopilotAPI(this.user.token)
+    try {
+      const relatedNotification = await this.db.clientNotification.findFirst({
```

Comment:
can't there be multiple client notifications related with the same task? 

```
[
  {
    clientId: 123,
    taskId: 321,
    notificationId: xxx,
  },
  {
    clientId: 123,
    taskId: 321,
    notificationId: xxx,
  },
]
```

it will mark the first notification as read?

Reply:
Yes, there are notifications triggered for other events as well like you said. However, this condition is avoided because the table only logs notifications that occur when the task is created.

Reply:
hmm ok.. in that case table name is not very clear.. may be add comments / rename.. as in future other devs might use this table for other purposes